### PR TITLE
jnp.where: provide fill_value for single-argument form

### PR DESCRIPTION
**Pull request by noahtren**
Extends #6910 & #7592. Padding with 0 can collide with actual zero indices and cause problems. We could specify the pad value, i.e. as -1.

### CODE REVIEW

**Review by jakevdp — APPROVED** [review]
Looks great, thanks!

File: jax/_src/numpy/lax_numpy.py [hunk 1]
```diff
@@ -1777,13 +1777,13 @@ def _where(condition, x=None, y=None):
...
 """
 
 @_wraps(np.where, update_doc=False, lax_description=_WHERE_DOC)
-def where(condition, x=None, y=None, *, size=None):
+def where(condition, x=None, y=None, *, size=None, fill_value=None):
   if x is None and y is None:
-    return nonzero(asarray(condition), size=size)
+    return nonzero(asarray(condition), size=size, fill_value=fill_value)
   else:
     if size is not None:
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:1788]: Perhaps check if `fill_value is not None` here as well?

Comment (noahtren) [reply jax/_src/numpy/lax_numpy.py:1791]: How's this? The semantics are the same but it only passes down `fill_value` if provided.

Comment (jakevdp) [reply jax/_src/numpy/lax_numpy.py:1791]: Sorry to be unclear: I meant below, where it raises an error `if size is not None`. We should change that to `if size is not None or fill_value is not None` and change the error text appropriately.

I think it's fine to pass `fill_value` to `nonzero` regardless of its value.

Comment (noahtren) [reply jax/_src/numpy/lax_numpy.py]: Got it!

**Review by jakevdp — COMMENTED** [review]
Looks great! The last thing we need is to squash all the changes into a single commit (see https://jax.readthedocs.io/en/latest/contributing.html#single-change-commits-and-pull-requests) Then we can get this merged!

**Review by jakevdp — APPROVED** [review]

**Comment by jakevdp** [comment]
Thanks for working on this!
